service/src/test/java/bio/terra/profile/pact/provider/BPMProviderTest.java

```diff
@@ -0,0 +1,137 @@
+import bio.terra.common.iam.AuthenticatedUserRequest;
+import bio.terra.common.iam.AuthenticatedUserRequestFactory;
+import bio.terra.profile.app.Main;
+import bio.terra.profile.app.configuration.BeanConfig;
+import bio.terra.profile.app.controller.UnauthenticatedApiController;
+import bio.terra.profile.db.ProfileDao;
+import bio.terra.profile.service.crl.AzureCrlService;
+import bio.terra.profile.service.crl.GcpCrlService;
+import bio.terra.profile.service.iam.SamService;
+import bio.terra.profile.service.iam.model.SamResourceType;
+import bio.terra.profile.service.job.JobService;
+import bio.terra.profile.service.profile.exception.ProfileNotFoundException;
+import bio.terra.profile.service.profile.model.BillingProfile;
+import bio.terra.profile.service.status.ProfileStatusService;
+import java.util.List;
+import java.util.Map;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.TestTemplate;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.boot.autoconfigure.data.jdbc.JdbcRepositoriesAutoConfiguration;
+import org.springframework.boot.autoconfigure.jdbc.DataSourceAutoConfiguration;
+import org.springframework.boot.context.properties.ConfigurationPropertiesScan;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.boot.test.web.server.LocalServerPort;
+import org.springframework.cache.CacheManager;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.FilterType;
+
+@Tag("provider-test")
+@Provider("bpm-provider")
+@PactBroker
+// for local testing, put any test pacts in the service/pacts folder.
+// then comment out the above line, and uncomment the following line
+// @PactFolder("pacts")
+@SpringBootTest(
+    webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT,
+    classes = Main.class,
+    properties = {
+      "profile.profile-database.initialize-on-start=false",
+      "profile.profile-database.upgrade-on-start=false"
+    })
+@EnableAutoConfiguration(
+    exclude = {DataSourceAutoConfiguration.class, JdbcRepositoriesAutoConfiguration.class})
+@ComponentScan(
+    basePackages = {"bio.terra.profile.app.controller", "bio.terra.profile.service"},
+    excludeFilters = {
+      @ComponentScan.Filter(
+          type = FilterType.ASSIGNABLE_TYPE,
+          value = {BeanConfig.class, ProfileDao.class}),
+      @ComponentScan.Filter(
+          type = FilterType.REGEX,
+          pattern = {"jdbcDialect", "jdbcCustomConversions", "jdbcMappingContext"})
+    })
+@ConfigurationPropertiesScan(basePackages = {"bio.terra.profile"})
+public class BPMProviderTest {
+
+  @LocalServerPort int port;
+
+  @MockBean AzureCrlService azureCrlService;
```

Comment:
At the current level of tests, most of mocks are just so spring has something to instantiate beans with when creating a context.
As we add tests, if we wanted to push the mocks lower down, we can: we just need to keep in mind that these tests are about testing the api-level interaction between BPM and consuming services, not about testing the internal logic of BPM.

Reply:
It would be worth putting a comment in the test that the mocks are only there so that spring can instantiate beans when creating a context.